CompetitionTemplate/src/competition/subsystems/arm/arm_commands/ArmAngleMaintainerCommand.java

```diff
@@ -33,11 +51,52 @@ public void initialize() {
 
     @Override
     public void execute() {
-        double currentArmAngle = armSubsystem.getArmAngle();
-        double targetArmAngle = armTargetSubsystem.getTargetAngle();
-
-        double armPower = pidManager.calculate(targetArmAngle, currentArmAngle);
-
-        armSubsystem.setArmMotorPower(armPower);
+        
+        // just in case the system underneath is not running
+        armSubsystem.updateSensors();
+        
+        if (armSubsystem.isCalibrated()) {
+        
+            double currentArmAngle = armSubsystem.getArmAngle();
+            double targetArmAngle = armTargetSubsystem.getTargetAngle();
+    
+            double armPower = pidManager.calculate(targetArmAngle, currentArmAngle);
+    
+            armSubsystem.setArmMotorPower(armPower);
+        }
+        else {
+            // System not calibrated! Let's try to fix that.
+            
+            if (!autoCalibrationEnabled.get()) {
```

Comment:
there's enough going on in here that maybe we should refactor this all into another method on the class, something like `attemptCalibration`